pyvo/utils/tests/test_prototype.py

```diff
@@ -0,0 +1,149 @@
+import warnings
+from copy import deepcopy
+from unittest import mock
+
+import pytest
+
+from pyvo.utils import prototype_feature, activate_features, prototype
+from pyvo.utils.prototype import PrototypeError, PrototypeWarning, Feature
+
+
+@pytest.fixture
+def prototype_function(features):
+    features({
+        'my-feature': Feature('my-feature', url='http://somewhere/else')
+    })
+
+    @prototype_feature('my-feature')
+    def i_am_prototype(arg):
+        arg('called')
+
+    return i_am_prototype
+
+
+@pytest.fixture
+def features():
+    previous_available = deepcopy(prototype.features)
+    prototype.features.clear()
+
+    def add(features):
+        prototype.features.update(features)
+
+    yield add
+
+    prototype.features.clear()
+    prototype.features.update(previous_available)
+
+
+def test_feature_turned_off_by_default(prototype_function):
+    with pytest.raises(PrototypeError) as e:
+        prototype_function(None)
+
+    assert str(e.value) == 'i_am_prototype is part of a prototype feature (my-feature) that has not been activated. ' \
+                           'For more information please visit http://somewhere/else'
+
+
+def test_activate_feature(prototype_function):
+    probe = mock.Mock()
+
+    activate_features('my-feature')
+
+    try:
+        prototype_function(probe)
+    except Exception as exc:
+        assert False, f"Should not have raised {exc}"
+
+    probe.assert_called_once_with('called')
+
+
+def test_non_existent_feature_warning():
+    with pytest.warns(PrototypeWarning) as w:
+        activate_features('i dont exist')
+
+    assert len(w) == 1
+    assert str(w[0].message) == 'No such feature "i dont exist"'
+
+
+def test_activate_all_features(features):
+    features({
+        'feat-one': Feature('feat-one'),
+        'feat-two': Feature('feat-two')
+    })
+
+    activate_features()
+
+    assert set(prototype.features.keys()) == {'feat-one', 'feat-two'}
+    assert prototype.features['feat-one'].on
+    assert prototype.features['feat-two'].on
+
+
+def test_decorate_class(features, recwarn):
+    features({
+        'class': Feature('class')
+    })
+    probe = mock.Mock()
+
+    @prototype_feature('class')
+    class FeatureClass:
```

Comment:
super nitpick: maybe use a more prototype like name. (It's a bit confusing that there is a `Feature` class, but that is for the inner working, and then there are the pyvo prototype features.

Reply:
Changing the name to `SomePrototype`